pkg/operator/ceph/object/admin.go

```diff
@@ -242,9 +242,12 @@ func runAdminCommand(c *Context, expectJSON bool, args ...string) (string, error
 	// simply because the external cluster mode does not support that yet
 	//
 	// The following conditions tries to determine if the cluster is external
-	// When connecting to an external cluster, the Ceph user is different than client.admin
-	// This is not perfect though since "client.admin" is somehow supported...
-	if c.Name != "" && c.clusterInfo.CephCred.Username == cephclient.AdminUsername {
+	// When connecting to an external cluster, the Ceph user is different than client.rookoperator
+	// This is not perfect though since "client.rookoperator" is somehow supported...
+
+	c.clusterInfo.CephCred.Username = cephclient.NonOperatorAdminUsername
```

Comment:
What was the error before you added this to override the username? I assume some commands were requiring the client.admin keyring explicitly instead of allowing client.rookoperator even though it has full privileges. I'm thinking we'll need the following approach:
- Both client.admin and client.rookoperator would be available (after mon quorum forms)
- The operator would need to generate the ceph.conf for both of those users whenever it starts, instead of just one of them
- The operator would perform most ceph calls with the rookoperator keyring, but still use client.admin where needed.

This likely will fix the remaining integration tests, though didn't get to dig deeply into the errors and implementation needs more thought.

Reply:
>What was the error before you added this to override the username? 

some commands were failing like rgw admin commands

Reply:
so, I fiqured that 4 commands require admin/or-failing

1.radosgw-admin
2. block pool deletion
3. filesystem deletion
4. mgr dashboard